[CodeGen] Resolve FIXME: Try SCEV getRange

AZero13 opened the pull request:
Using the range to see if an expression is guaranteed to be a safe entry rather than only accepting a constant value allows us to examine more expressions.

dtcxzyw commented:
Tests?

nikic commented on llvm/lib/CodeGen/SafeStack.cpp:
```diff
@@ -263,10 +263,16 @@ bool SafeStack::IsMemIntrinsicSafe(const MemIntrinsic *MI, const Use &U,
-  return IsAccessSafe(U, Len->getZExtValue(), AllocaPtr, AllocaSize);
+  const SCEV *LenSCEV = SE.getSCEV(MI->getLength());
+  ConstantRange LenRange = SE.getUnsignedRange(LenSCEV);
+  auto element = LenRange.getSingleElement();
```
There's no point to doing this if you're only going to handle single-element ranges.

AZero13 committed:
[CodeGen] Resolve FIXME: Try SCEV getRange

Using the range to see if an expression is guaranteed to be a safe entry rather than only accepting a constant value allows us to examine more expressions.

nikic reviewed (changes requested):
No tests
